components/UpdateEmailForm/UpdateEmailForm.tsx

```diff
@@ -0,0 +1,181 @@
+import { AlertVariants } from '@/components/AlertNotification/Alerts.enum';
+import { Button } from '@/components/Button/Button';
+import { Input } from '@/components/Input/Input';
+import LoadingSpinner from '@/components/LoadingSpinner/LoadingSpinner';
+import { useDataStore } from '@/store/dataStore';
+import { zodResolver } from '@hookform/resolvers/zod';
+import { JSX, useState } from 'react';
+import { Control, SubmitHandler, useForm, useWatch } from 'react-hook-form';
+import toast from 'react-hot-toast';
+import { z } from 'zod';
+import {
+  Form,
+  FormControl,
+  FormField,
+  FormItem,
+  FormLabel,
+  FormMessage,
+} from '../Form/Form';
+
+const UpdateEmailSchema = z.object({
+  email: z
+    .string()
+    .min(1, { message: 'Please enter an email address' })
+    .email({ message: 'Please enter a valid email address' }),
+  password: z.string().min(1, { message: 'Please enter a password' }),
+});
+
+type UpdateEmailSchemaType = z.infer<typeof UpdateEmailSchema>;
+
+/**
+ * Display update user email form
+ * @returns {JSX.Element} The rendered update user email form.
+ */
+const UpdateEmailForm = (): JSX.Element => {
+  const [isUpdating, setIsUpdating] = useState<boolean>(false);
+  const { user, updateUser } = useDataStore((state) => state);
+
+  const form = useForm<UpdateEmailSchemaType>({
+    resolver: zodResolver(UpdateEmailSchema),
+    defaultValues: {
+      email: user.email,
+      password: '',
+    },
+  });
+
+  const email: string = useWatch({
+    control: form.control,
+    name: 'email',
+  });
+
+  const password: string = useWatch({
+    control: form.control,
+    name: 'password',
+  });
+
+  /**
+   * A function that handles form submission.
+   * @param {UpdateEmailSchemaType} data - The data submitted in the form.
+   * @returns {Promise<void>} Promise that resolves after form submission is processed.
+   */
+  const onSubmit: SubmitHandler<UpdateEmailSchemaType> = async (
+    data: UpdateEmailSchemaType,
+  ): Promise<void> => {
+    const { email, password } = data;
+    setIsUpdating(true);
+    try {
+      await updateUserEmail({
+        email,
+        password,
+      });
+
+      toast.custom(
+        <Alert
+          variant={AlertVariants.Success}
+          message="You have successfully updated your email."
+        />,
+      );
+
+      updateUser(user.documentId, user.id, email, user.leagues);
+      form.reset({ email: email || '', password: '' });
+    } catch (error) {
+      console.error('Email Update Failed', error);
+      toast.custom(
+        <Alert variant={AlertVariants.Error} message="Email Update Failed!" />,
+      );
+    } finally {
+      setIsUpdating(false);
+    }
+  };
+
+  const isDisabled = email === user.email || !email || isUpdating || !password;
+
+  return (
+    <Form {...form}>
+      <form
+        id="email-input-container"
+        className="border border-border rounded-lg w-full bg-border/40 p-6"
```

Comment:
Does this need `bg-border/40`? Do none of the existing CSS variables work for it?

Reply:
@ryandotfurrer I didn't try any of the other ones.  This is a unique background color than what we have used so I wanted it to be same color as border but a little less bright using opacity.